crates/ruff_python_parser/src/lexer.rs

```diff
@@ -1236,6 +1220,49 @@ const fn is_python_whitespace(c: char) -> bool {
     )
 }
 
+enum LexedText<'a> {
```

Comment:
is that the same abstraction we use for normalizing strings?

Reply:
The lexer doesn't normalize Strings. I haven't looked into what the whole `string.rs` is doing (and how much of it could be moved into the lexer)

Reply:
i meant in the formatter, this was just a random thought because the logic looked familiar from the normalize functions of strings, ints and floats

Reply:
It's similar but we don't use an abstraction in the formatter.